src/main/java/com/team4/testingsystem/services/impl/ChosenOptionServiceImpl.java

```diff
@@ -33,16 +39,22 @@ public ChosenOption getByTestAndQuestionId(Long testId, Long questionId) {
     }
 
     @Override
-    public void save(ChosenOption chosenOption) {
-        try {
-            chosenOptionRepository.save(chosenOption);
-        } catch (EntityNotFoundException exception) {
-            throw new ChosenOptionBadRequestException();
+    public void saveAll(List<ChosenOption> chosenOptions) {
+        Test test;
+
+        Question question;
+
+        for (ChosenOption item : chosenOptions) {
+            test = item.getId().getTest();
+            question = item.getId().getQuestion();
+
+            restrictionsService.checkOwnerIsCurrentUser(test);
+
+            restrictionsService.checkStartedStatus(test);
+
+            restrictionsService.checkTestContainsQuestion(test, question);
         }
-    }
```

Comment:
`chosenOptionRepository.saveAll(chosenOptions);`?

Reply:
> `chosenOptionRepository.saveAll(chosenOptions);`?

Full method: 
![image](https://user-images.githubusercontent.com/71790626/129879391-ddf9e6e9-4488-4589-9906-6965da81e9dd.png)

Reply:
And what do you think about parallel stream here?